packages/react/src/composite/list/useCompositeListItem.ts

```diff
@@ -33,13 +34,23 @@ export function useCompositeListItem<Metadata>(
 
       if (index !== null && node !== null) {
         elementsRef.current[index] = node;
+
         if (labelsRef) {
           const isLabelDefined = label !== undefined;
-          labelsRef.current[index] = isLabelDefined ? label : (node?.textContent ?? null);
+          let textContent = node.textContent;
```

Comment:
Don't call `.textContent` if the value will be replaced by `getItemText` return value.

Reply:
It might not necessarily be replaced if it's nullish